### PR TITLE
Added platform-name wildcard support

### PR DESCRIPTION
**Pull request by yarick**
Added platform-name wildcard support
Added unit test for platform-name wildcard
Added doc for release wildcard
Added doc for platform-name wildcard

Signed-off-by: superyarick <yarick@yarick.net>

## Description
Added a function to convert wildcartds in platform-name to match against regex - borrowed from existing code for matching 'release's

### Note
The current case statement that evaluates the ```supports``` statements does not allow for usage of both ```platform, release and platform-name``` in conjunction as it is a fast failing case statement. I will open a follow-on issue with respect to this.
https://github.com/inspec/inspec/blob/master/lib/inspec/resources/platform.rb#L76-L88

## Related Issue
Did not open an issue. Just submitted small fix.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by aaronlippold** [comment]
Given this was a clean fork of inspec as of today - now sure why we are getting the ruby-2.4 issues

**Comment by Schwad** [comment]
@aaronlippold I'm wondering if there was a bump of the parallel gem which killed 2.4. I'll raise some issues around this

`Gem::RuntimeRequirementNotMetError: parallel requires Ruby version >= 2.5`

**Comment by aaronlippold** [comment]
@Schwad so can we still move ahead with peer review on the code and docs while that parallel issue is being resolved with the backend team?

**Comment by Schwad** [comment]
@aaronlippold 
> so can we still move ahead with peer review on the code and docs while that parallel issue is being resolved with the backend team?

Yes, peer review may continue whilst this is being resolved.

**Comment by Schwad** [comment]
@yarick @aaronlippold we have merged the parallel fix so you can rebase and this should pass now 💯

**Comment by aaronlippold** [comment]
Cool, guess I am teaching rebasing today :) hold my bear...
--------
Aaron Lippold

lippold@gmail.com

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

**Comment by aaronlippold** [comment]
new updates from maser merged in.

**Comment by aaronlippold** [comment]
Although this works - we do have a small bit of unexpected or at least unclear behavior - which we documented as a side step - but the situation I would like to clean up as a follow on would be.

The case statement could be made to support more logical filtering in its organization.  As you can see from the note, if we do something like:
```
supports:
  - platform-family: windows
  - platform-name: windows-server-2016*
```
we pass fast on the case statement just be happy it got `windows` back and support 2019 or 2012 etc

As documented in this PR - if you want to filter on a specific sub-family by name you need to just use:
```
supports:
  - platform-name: windows-server-2016* 
```

I think however the real user community would expect a behavior more along the lines of:
```
supports:
  - platform-family: windows
    name: *2016*
```
or, even a bit more detail, given that both 2016 and 2019 have a family of releases in the 10.0.1* range
```
supports:
  - platform-family: windows
    name: *2019*
    release: 10.0.1.17*
```

**Comment by aaronlippold** [comment]
The use case here is that I want a wrapper profile that I run via a lambda or ssm function that just has all my os baselines for windows and or linux and just runs the right one for the right platform ...

**Comment by aaronlippold** [comment]
@Schwad what's up with buildkit?

**Comment by aaronlippold** [comment]
Hi all, are we good to go on merging this? Thanks.

File: docs-chef-io/content/inspec/profiles.md [hunk 1]
```diff
@@ -142,6 +142,15 @@ supports:
     release: 14.04
 ```
 
+and to target entire release of Ubuntu version 14.x
```
Comment (mjingle) [inline docs-chef-io/content/inspec/profiles.md]:
```suggestion
To target the entire release of Ubuntu version 14.x, use:
```

File: docs-chef-io/content/inspec/profiles.md [hunk 2]
```diff
@@ -150,6 +159,14 @@ supports:
   - platform-family: redhat
 ```
 
+and to target the entire Windows 2019 platform family (including Datcenter and Core Servers):
```
Comment (mjingle) [inline docs-chef-io/content/inspec/profiles.md]:
```suggestion
To target the entire Windows 2019 platform family including Datcenter and Core Servers, use:
```

File: docs-chef-io/content/inspec/profiles.md [hunk 3]
```diff
@@ -170,6 +187,7 @@ supports:
   - platform: aws
 ```
 
+
```
Comment (mjingle) [inline docs-chef-io/content/inspec/profiles.md]:
```suggestion
```

Comment (mjingle) [reply docs-chef-io/content/inspec/profiles.md]: A single blank line is needed between the yaml code and the header. Line 189 fulfills this requirement already.

File: docs-chef-io/content/inspec/profiles.md [hunk 4]
```diff
@@ -115,8 +115,8 @@ inspec check examples/profile
 Use the `supports` setting in the `inspec.yml` file to specify one (or more) platforms for which a profile is targeting. The list of supported platforms may contain the following:
 
 - Use `platform-family` to restrict to a specific platform family.
-- Use `platform-name` to restrict on a specific platform name.
-- Use `release` to restrict to a specific platform version (used with platform-name).
+- Use `platform-name` to restrict on a specific platform name (supports bash wildcards).
+- Use `release` to restrict to a specific platform version (used with platform-name, supports bash wildcards).
```
Comment (mjingle) [inline docs-chef-io/content/inspec/profiles.md]:
```suggestion
- Use `release` to restrict to a specific platform version, and use together with `platform-name`. `release` supports asterisk (`*`) wildcard use.
```

File: docs-chef-io/content/inspec/profiles.md [hunk 5]
```diff
@@ -115,8 +115,8 @@ inspec check examples/profile
 Use the `supports` setting in the `inspec.yml` file to specify one (or more) platforms for which a profile is targeting. The list of supported platforms may contain the following:
 
 - Use `platform-family` to restrict to a specific platform family.
-- Use `platform-name` to restrict on a specific platform name.
-- Use `release` to restrict to a specific platform version (used with platform-name).
+- Use `platform-name` to restrict on a specific platform name (supports bash wildcards).
```
Comment (mjingle) [inline docs-chef-io/content/inspec/profiles.md]:
```suggestion
- Use `platform-name` to restrict on a specific platform name. `platform-name` supports asterisk (`*`) wildcard use.
```

**Comment by mjingle** [comment]
@yarick Hello! Thank you for updating the documentation with your code. Please incorporate my suggested documentation edits, and then you will good to go from a docs perspective. Thank you in advance and let me know if you have any questions.

**Comment by yarick** [comment]
@mjingle - I believe I have addressed all your comments.  Let me know if there is anything else.

**Review by mjingle — APPROVED** [review]
@yarick Your docs update looks great! Thank you so much. You are good to go from a docs perspective. 👍

**Review by clintoncwolfe — APPROVED** [review]
Thanks so much!

**Comment by Schwad** [comment]
@aaronlippold @yarick thanks much for this! Merging now. 

In future please squash your PRs into one commit (unless there's a good reason from a documentation standpoint not to)

**Pull request merged**